activejob/lib/active_job/test_helper.rb

```diff
@@ -239,6 +241,14 @@ def enqueued_jobs_size(only: nil)
             enqueued_jobs.size
           end
         end
+
+        def serialize_args_for_assertion(args)
+          original_args = args.dup
+          if job_args = args.delete(:args)
+            args[:args] = ActiveJob::Arguments.serialize(job_args)
+          end
+          original_args
```

Comment:
Probably a little weird but I need both the original and serialized args. The serialized args are used in the assertion comparison and the original args are used in the failure message if the assertion fails. It probably makes immediate sense if you place that code in the assertion method as follows:

```
def assert_enqueued_with(args = {}, &_block)
  original_enqueued_jobs = enqueued_jobs.dup
  clear_enqueued_jobs

  args.assert_valid_keys(:job, :args, :at, :queue)
  original_args = args.dup
  if job_args = args.delete(:args)
    args[:args] = ActiveJob::Arguments.serialize(job_args)
  end

  yield
  matching_job = enqueued_jobs.any? do |job|
    args.all? { |key, value| value == job[key] }
  end

  assert matching_job, "No enqueued job found with #{original_args}"
ensure
  queue_adapter.enqueued_jobs = original_enqueued_jobs + enqueued_jobs
end
```

However I would have to repeat it in ```assert_performed_with``` so I decided to place it in a helper method. Would you prefer I remove the helper in favour of the above?